src/posix/client.cpp

```diff
@@ -331,7 +331,7 @@ int main(int argc, char *argv[])
                 if (lineStart < rval && promptState != 1)
                 {
                     assert(promptState != 0 && promptState != 2);
-                    VerifyOrExit(DoWrite(STDOUT_FILENO, buffer + lineStart, rval - lineStart), ret = OT_EXIT_FAILURE);
+                    VerifyOrExit(DoWrite(STDOUT_FILENO, buffer + lineStart, (size_t)(rval - lineStart)), ret = OT_EXIT_FAILURE);
```

Comment:
Prefer C++-style casts:
```suggestion
                    VerifyOrExit(DoWrite(STDOUT_FILENO, buffer + lineStart, static_cast<size_t>(rval - lineStart)), ret = OT_EXIT_FAILURE);
```